### DISCUSSION TITLE
我在重新整理kernel代码

### DISCUSSION THREAD

**Discussion by zbwu** [MiRoboticsLab/cyberdog_ros2#23] [Ideas]
这两天在重新整理Cyberdog的kernel代码，主要包括以下几点

以nv-tegra.nvidia.com上的代码库为基础，包含完整的历史，可以方便以后的版本升级和整合

为Cyberdog创建独立dts文件，避免和Developer Kit共用产生的一些冲突

更新OV7251(双目相机)和OV13B10(彩色相机)的驱动，支持不同的分辨率和帧率

========

这个是我学习Cyberdog的一部分，之后是构建整个固件（NX部分），然后是用户空间的视频处理

**Comment by homalozoa** [comment]
感谢支持铁蛋项目，欢迎提供PR！十一后我会把内核代码迁移到GitHub。

**Comment by zbwu** [comment]
此时的想法是…

感觉做BSP的同学没有看文档…

**Reply by homalozoa** [reply]
基本上要重新弄一遍了。。

**Reply by liseri** [reply]
> 基本上要重新弄一遍了。。

希望铁蛋的开源，可以先从nvida jetpack初始化环境开始，既不是小米的自定义镜像，也不需要一开始就需要动内核代码

**Comment by zbwu** [comment]
本地基本完成了当初的计划，但还没有充分测试。因为官方开源计划的变化，这个idea暂停一下。

**Reply by liseri** [reply]
开源计划怎么变化了，我想从官方jetpack初始化环境开始，能搭建完成铁蛋功能；

**Reply by homalozoa** [reply]
有一部分硬件强相关的内容还没有完全解耦开，还在逐步拆。

**Reply by zbwu** [reply]
> 开源计划怎么变化了，我想从官方jetpack初始化环境开始，能搭建完成铁蛋功能；

运动控制部分官方不开源了。目前开源的部分已经足够从jetpack重建整个系统

**Reply by liseri** [reply]
> 有一部分硬件强相关的内容还没有完全解耦开，还在逐步拆。

你好，能麻烦帮忙看个问题么，实在找不到好的地方了，圈子不活跃；
问题是 刷机后 蓝牙连接失败，之前刷过一次小米镜像，没出问题，我今天刷了一次Jetpack官方镜像，又刷回小米镜像，发现手机app连不上蓝牙了， bluetooth_ros2.service报错如下：
Nov 08 02:53:57 lubuntu ros2[17548]: 2021-11-08 02:53:57,015 - gattserver.py[line:1041] - DEBUG: main: set bleStatus BLE_STATUS_NONE -> BLE_STATUS_ADV
Nov 08 02:53:57 lubuntu ros2[17548]: 2021-11-08 02:53:57,016 - gattserver.py[line:950] - INFO: RegADV: RegisterAdvertisement
Nov 08 02:53:57 lubuntu ros2[17548]: 2021-11-08 02:53:57,017 - gattserver.py[line:962] - INFO: RegATT: RegisterApplication
Nov 08 02:53:57 lubuntu ros2[17548]: 2021-11-08 02:53:57,017 - gattserver.py[line:253] - INFO: get_path: <bluetooth.gattserver.RobotApplication at / at 0x7f9ae12518>
Nov 08 02:53:57 lubuntu ros2[17548]: 2021-11-08 02:53:57,020 - gattserver.py[line:1045] - DEBUG: main init and send audio cmd BOOT_COMPLETE
Nov 08 02:53:57 lubuntu ros2[17548]: 2021-11-08 02:53:57,024 - gattserver.py[line:1048] - DEBUG: RegADV: BT enable adv, set LED_BT_ADV
Nov 08 02:53:57 lubuntu ros2[17548]: 2021-11-08 02:53:57,040 - gattserver.py[line:783] - INFO: register_app_cb: GATT application registered!!!
Nov 08 02:53:57 lubuntu ros2[17548]: 2021-11-08 02:53:57,041 - gattserver.py[line:198] - INFO: GetAll
Nov 08 02:53:57 lubuntu ros2[17548]: 2021-11-08 02:53:57,041 - gattserver.py[line:201] - INFO: returning props
Nov 08 02:53:57 lubuntu ros2[17548]: **ERROR:dbus.service:Unable to append ({'Type': 'peripheral', 'ServiceUUIDs': dbus.Array(['C420'], signature=dbus.Signature('s')), 'ManufacturerData': dbus.Dictionary({65535: dbus.Array([0, 1, 2, 3, 4], signature=dbus.Signature('y'))}, signature=dbus.Signature('qv')), 'ServiceData': dbus.Dictionary({'9999': dbus.Array([0, 1, 2, 3, 4], signature=dbus.Signature('y'))}, signature=dbus.Signature('sv')), 'LocalName': dbus.String('铁蛋ne QED_ROCE_MODIFY\nne QED_ROCE_MODIFY\nne QED_ROCE_MODIFY\nne QED_ROCE_MODIFY\n_MDS_OP_LOOKUPSNAP\n\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\nne _NET_PSNAP_H\nne HFI1_CAP_EXTEND\nne  PCI_EXP_SLTCAP\nef void (APIENTRYP\nef Bool ( *PFNGLXQ\nef void (GLAPIENTR\nFUN_EXPORT PFNGLEV\nest.Adobe-DPS-Exte\nest.Adobe-DPS-Exte\n'), 'IncludeTxPower': dbus.Boolean(True)},) to message with signature a{sv}: <class 'ValueError'>: embedded null byte**
Nov 08 02:53:57 lubuntu ros2[17548]: 2021-11-08 02:53:57,048 - gattserver.py[line:794] - INFO: register_ad_cb: Advertisement registered!!!
Nov 08 02:53:57 lubuntu ros2[17548]: INFO:/var/log/bluetooth.log:register_ad_cb: Advertisement registered!!!
麻烦了！！

**Reply by homalozoa** [reply]
硬件平台是铁蛋还是NV开发板？ 可以提个issue
